javacutil/src/org/checkerframework/javacutil/InternalUtils.java

```diff
@@ -349,7 +349,9 @@ public static TypeMirror greatestLowerBound(
         if (t2.getKind() == TypeKind.WILDCARD) {
             return t1;
         }
-        return types.glb(t1, t2);
+        // call glb on unannotatedType, otherwise glb may cause a stack overflow error
+        // See Issue #500
+        return types.glb(t1.unannotatedType(), t2.unannotatedType());
```

Comment:
The code for leastUpperBound a bit further up in this class looks very similar.
Should we add the same calls to unannotatedType up there, even if we don't have a test case for it?